### PR TITLE
Python: fix lock file updates

### PR DESCRIPTION
**Pull request by helderco**
Fixes https://github.com/dagger/dagger/issues/7033

To simplify not needing to install the SDK's dependencies for codegen, the Python codegen script has been moved into an independent package with a minimal dependency.

Added a `codegen introspect` subcommand to avoid the need for the Python codegen to have HTTP capabilities in order to introspect from the API directly.

This helped simplifiy the runtime module. Now, if a module has a `requirements.lock` file, it'll be re-compiled automatically (without upgrading packages). Without this, changes to the SDK's dependencies in a way that affects `import dagger`, would fail during `dagger develop`, leading to a difficult manual update of `requirements.lock`.

### CODE REVIEW

File: cmd/codegen/main.go
Comment (wingyplus) [inline cmd/codegen/main.go:86]: That's nice!!

Comment (helderco) [reply cmd/codegen/main.go:86]: Yup 🙂 At some point it can come from the codegen module, but this was a quicker fix for now and can be reused in other SDKs.

Comment (wingyplus) [reply cmd/codegen/main.go:86]: I have plan to use the introspection from that module. But after see this PR, I will wait until this PR merge then. 😂

**Comment by helderco** [comment]
Getting lots of failures due to:
```
429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit",
```

**Comment by jedevc** [comment]
> Getting lots of failures due to:
> 
> ```
> 429 Too Many Requests - Server message: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit",
> ```

Should be fixed by https://github.com/dagger/dagger/pull/7072, sorry for the inconvenience!

File: ci/sdk_python.go
Comment (jedevc) [inline ci/sdk_python.go:22]: :eyes:

**Review by jedevc — APPROVED** [review]

File: ci/sdk_python.go
Comment (helderco) [reply ci/sdk_python.go:22]: I added this in a first commit here, but then split in another PR. After rebasing looks like the git thinks I want to remove it!

**Pull request merged**
